### PR TITLE
test(go,java,node): convert yoshi-go, node tests to completely use sinon

### PR DESCRIPTION
**Pull request by chingor13**
GitHub class/tests should be responsible for making the correct API calls. The releasers should only need to handle the outputs from the GitHub methods.

* Adds test helper for building mock commits for stubbing `GitHub#commitsSinceSha`.
* Adds test helper for building `GitHubFileContents` for `GitHub#getFileContentsOnBranch`
* Refactors yoshi-go tests to mock `GitHub` rather than mocking the low level http requests with nock

### CODE REVIEW

**Review by codyoss — APPROVED** [review]
LGTM. I appreciate not having to manually manipulate JSON 👏🏻

File: test/releasers/utils.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GitHubFileContents} from '../../src/github';
+import {Commit} from '../../src/graphql-to-commits';
+import {readFileSync} from 'fs';
+import {resolve} from 'path';
+import * as crypto from 'crypto';
+
+export function buildGitHubFileContent(
```
Comment (joeldodge79) [inline test/releasers/utils.ts:21]: thank you! I'd done this on my aggregate branch too :-)

**Review by joeldodge79 — APPROVED** [review]
👍

File: test/releasers/yoshi-go.ts [hunk 2]
```diff
@@ -47,71 +53,60 @@ describe('YoshiGo', () => {
...
+      );
+      getFileContentsStub.rejects(
+        Object.assign(Error('not found'), {status: 404})
+      );
+
+      sandbox
+        .stub(releasePR.gh, 'commitsSinceSha')
+        .resolves([
+          buildMockCommit('fix(automl): fixed a really bad bug'),
+          buildMockCommit('feat(asset): added a really cool feature'),
```
Comment (bcoe) [inline test/releasers/yoshi-go.ts:89]: I love how much better this reads.

**Review by bcoe — APPROVED** [review]

**Pull request merged**